src/foam/java/refinements.js

```diff
@@ -435,10 +435,10 @@ foam.CLASS({
       var m = info.getMethod('cast');
       m.body = 'return ( o instanceof Number ) ?'
             + '((Number)o).intValue() :'
-            + '(int)o;';
+            + '((Integer) o).intValue();';
```

Comment:
It wouldn't, however the compiler throws an error for attempting to cast a generic object to a primitive.